### PR TITLE
[INLONG-10093][SDK] Support to transform from CSV protocol to CSV/KV protocol by single SQL

### LINKED ISSUE #10093: [Feature][SDK] Support to transform from CSV/KV protocol to CSV/KV protocol by single SQL
### Description

Support to transform from CSV protocol to CSV/KV protocol by single SQL

### Use case

Support to transform from CSV protocol to CSV/KV protocol by single SQL

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luchunliang**
Fixes #10093 

Parent issue: #10023 

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/TransformProcessor.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    private SourceDecoder decoder;
+    private SinkEncoder encoder;
+    private Charset srcCharset = Charset.defaultCharset();
+    protected Charset sinkCharset = Charset.defaultCharset();
+
+    private PlainSelect transformSelect;
+    private ExpressionOperator where;
+    private Map<String, ValueParser> selectItemMap;
+
+    public TransformProcessor(TransformConfig config) throws JSQLParserException {
```
Comment (justinwwhuang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/TransformProcessor.java:72]: Does this mean that the user needs to construct the TransformConfig properly? Can it be changed to passing in a JSON string

Comment (luchunliang) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/TransformProcessor.java:83]: Fixed

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/TransformProcessor.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                }
+            }
+        }
+    }
+
+    public List<String> transform(byte[] srcBytes, Map<String, Object> extParams) {
+        SourceData sourceData = this.decoder.decode(srcBytes, extParams);
+        List<String> sinkDatas = new ArrayList<>(sourceData.getRowCount());
+        for (int i = 1; i <= sourceData.getRowCount(); i++) {
+            if (!this.where.check(sourceData, i)) {
```
Comment (justinwwhuang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/TransformProcessor.java:143]: 'where' maybe null here

Comment (luchunliang) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/TransformProcessor.java]: Fixed

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/decode/KvSourceDecoder.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Override
+    public SourceData decode(byte[] srcBytes, Map<String, Object> extParams) {
+        String srcString = new String(srcBytes, srcCharset);
+        return this.decode(srcString, extParams);
+    }
+
+    @Override
+    public SourceData decode(String srcString, Map<String, Object> extParams) {
+        List<Map<String, String>> rowValues = KvUtils.splitKv(srcString, '&', '=', '\\', '\"', '\n');
```
Comment (justinwwhuang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/decode/KvSourceDecoder.java:55]: Are these characters fixed?

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/decode/SplitUtils.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        fields.add(new String(fieldValue, 0, fieldIndex));
+        fieldIndex = 0;
+        lines.add(fields.toArray(new String[0]));
+
+        String[][] result = new String[lines.size()][];
+        for (int i = 0; i < lines.size(); i++) {
+            result[i] = lines.get(i);
+        }
+        return result;
+        // switch (state) {
```
Comment (justinwwhuang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/decode/SplitUtils.java:169]: Unuseful code, please delete it

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/decode/KvSourceDecoder.java (hunk 3 above)
Comment (luchunliang) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/decode/KvSourceDecoder.java:55]: It is fixed for KV format.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/decode/SplitUtils.java (hunk 4 above)
Comment (luchunliang) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/decode/SplitUtils.java]: Removed.

**Review by justinwwhuang — APPROVED** [review]
LGTM

**Review by vernedeng — APPROVED** [review]
LGTM

**Pull request merged**